src/modules/network_manager/network_params.c

```diff
@@ -0,0 +1,180 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file network_params.c
+ * Parameter definition for network.
+ *
+ */
+
+
+/**
+ * Board Unique MAC Address byte #0
+ * Address is read by MAC EEPROM from the board and should be not modified by user
+ *
+ * @group Ethernet
+ * @min 0
+ * @max 255
+ */
+PARAM_DEFINE_INT32(ETH_MAC_ADDR_0, 0);
+
+/**
+ * Board Unique MAC Address byte #1
+ * Address is read by MAC EEPROM from the board and should be not modified by user
+ *
+ * @group Ethernet
+ * @min 0
+ * @max 255
+ */
+PARAM_DEFINE_INT32(ETH_MAC_ADDR_1, 0);
+
+/**
+ * Board Unique MAC Address byte #2
+ * Address is read by MAC EEPROM from the board and should be not modified by user
+ *
+ * @group Ethernet
+ * @min 0
+ * @max 255
+ */
+PARAM_DEFINE_INT32(ETH_MAC_ADDR_2, 0);
+
+/**
+ * Board Unique MAC Address byte #3
+ * Address is read by MAC EEPROM from the board and should be not modified by user
+ *
+ * @group Ethernet
+ * @min 0
+ * @max 255
+ */
+PARAM_DEFINE_INT32(ETH_MAC_ADDR_3, 0);
+
+/**
+ * Board Unique MAC Address byte #4
+ * Address is read by MAC EEPROM from the board and should be not modified by user
+ *
+ * @group Ethernet
+ * @min 0
+ * @max 255
+ */
+PARAM_DEFINE_INT32(ETH_MAC_ADDR_4, 0);
+
+/**
+ * Board Unique MAC Address byte #5
+ * Address is read by MAC EEPROM from the board and should be not modified by user
+ *
+ * @group Ethernet
+ * @min 0
+ * @max 255
+ */
+PARAM_DEFINE_INT32(ETH_MAC_ADDR_5, 0);
+
+/**
+ * Board IP Address byte #0
+ *
+ * @group Ethernet
+ * @min 0
+ * @max 255
+ */
+PARAM_DEFINE_INT32(ETH_IP_ADDR_0, 192);
```

Comment:
> the fact that one parameter can be a very long number and it seemed like less comfortable solution.

Yes exactly. That's why we should to extend QGC so that it shows the IP in the form `192.168.1.1`.

Reply:
:+1:

Reply:
Well, there's a problem doing so...
Let's say we wan't to change the IP parameter value to 192.168.0.1 with the param set command.
The actual value that will be stored is 127.255.255.255 (LONG_MAX), I suspect the problem is with the do_set function that uses strtol.

Reply:
I may be missing something here. see https://www.ipaddressguide.com/ip  The idea is the data type is IP_ADDRESS. The code on the GUI understands how to get the number and convert it to a long. The code on the platform (FMU) will be able to use the binary long format

Reply:
Yes, you are right.
But let's say we want to change the parameter with "param set ETH_IP_ADDR 3232235521" (3232235521 - 192.168.0.1), we get to do_set function with 3232235521, but inside do_set there's a call to strtol.
strtol will return 2147483647 (LONG_MAX) instead -1062731775 because of [this](https://github.com/PX4-NuttX/nuttx/blob/8957a027f4214416646f384bb47b838ec3686643/libc/stdlib/lib_strtol.c#L116), so the binary is now different.

Reply:
Oh! So we would need to fix the parameters. But that would cause more bloat to go to 64 bits (unless the union already is that big - IDNR) 

I think this further reinforces @dagar 
https://github.com/PX4/Firmware/pull/10124#issuecomment-410824212 and also suggest the parameters need some design effort for isolation (things stores but not user facing) and scalability.

Reply:
> We should then make sure QGC learns how to display IP addresses (@DonLakeFlyer @dogmaphobic ) and add the metadata.

That would require a parameter type of IP address to be added to mavlink and then flow through to QGC changes.

Reply:
> why not mac using 2 parameters?

We can do that too but it's more than 4 bytes. A 64 bit param type would be helpful.

For the `param set` we can easily extend it to accept the form `x.y.z.w`.

--- 
@DonLakeFlyer 
> That would require a parameter type of IP address to be added to mavlink and then flow through to QGC changes.

Why mavlink? We can do it through the unit in the param xml metadata. It would still be an INT32 as underlying type.

Reply:
> Oh! So we would need to fix the parameters. But that would cause more bloat to go to 64 bits

Why not just using uint32? why 64?

> For the param set we can easily extend it to accept the form x.y.z.w.

I think this is my preferred idea, much more user friendly

> We can do that too but it's more than 4 bytes. A 64 bit param type would be helpful.

Yes, it would be. But if for some reason we don't want a 64 bit param we can use 2 32 bit param for mac, do you think it's a good idea? or is it just another thing that will complicate the user experience?